### ISSUE TITLE
[New Feature] Purchase cycle for Product Bundle

### ISSUE THREAD

**Issue by GSLabIt** [frappe/erpnext#11475]
Hi all, 

it seems to me a **_important feature to be implemented_**. 

Having purchase receipt for an `Item`, and this item is a `Product Bundle`, it should **increase** stock for all the `items` included in the `Product Bundle` as per `qty` defined in it.

Let's have an example, product bundle include:

<img width="966" alt="schermata 2017-11-06 alle 20 29 33" src="https://user-images.githubusercontent.com/11652845/32460705-e2f3c93e-c333-11e7-98ca-e377a40f8850.png">

It's expected that for a purchase receipt of a `Product Bundle`,  i.e. `qty = 1`, stock of`Product Bundle Item` should be updated accordingly.

Basically is a purchase cycle feature for Product Bundle.

I'm willing to contribute for the development.

**Comment by elvise** [comment]
Here we have the same problem

**Comment by bhavishyasharma** [comment]
Hi everyone!

I have implemented the bundle in Buying Cycle. The valuation of child items is calculated based on the weightage which is defined in the ProductBundle

Steps:

1. Add weightage field to Product Bundle.
![image](https://user-images.githubusercontent.com/15744297/58901508-f32b9880-871e-11e9-8178-165c60f1dac6.png)
2. Add weightage field to Packed Item.
3. Add rate and amount fields to Packed Item.
![image](https://user-images.githubusercontent.com/15744297/58902099-1571e600-8720-11e9-9ecd-6b67fcbe0be5.png)

4. Add packed_items to Purchase Order, Purchase Receipt, Purchase Invoice.
<img width="1322" alt="Screenshot 2019-06-04 at 23 18 59" src="https://user-images.githubusercontent.com/15744297/58901743-68976900-871f-11e9-88b1-f745c35f96da.png">

5. Pass stock_entry for Purchase Receipt and Purchase Invoice.

Please let me know how to push the code adhering to your development guidelines.

**Comment by federicocalvo** [comment]
@bhavishyasharma Hi! is a very required fucnionality, someone contacted you for the PR? Do you have the link to share it or even our programmers try to do the PR with frappe? Thanks!

@GSLabIt @Tropicalrambler hi guys, did any of you see the message with @bhavishyasharma input? or did you solve it in some other way?

**Comment by yadavyk** [comment]
[#27124](https://github.com/frappe/erpnext/pull/27124)
